packages/manager/apps/key-management-service/src/data/hooks/useIdentity.ts

```diff
@@ -0,0 +1,193 @@
+import { UseQueryOptions, useQueries, useQuery } from '@tanstack/react-query';
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import {
+  getIdentityUsersIdsQueryKey,
+  getIdentityUsersIds,
+  getIdentityUserQueryKey,
+  getIdentityUser,
+  getIdentityGroupsIdsQueryKey,
+  getIdentityGroupsIds,
+  getIdentityGroup,
+  getIdentityGroupQueryKey,
+  getIdentityServiceAccountQueryKey,
+  getIdentityServiceAccount,
+  getIdentityServiceAccountsIdsQueryKey,
+  getIdentityServiceAccountsIds,
+} from '../api/identity';
+import {
+  Identity,
+  IdentityGroup,
+  IdentityOauthClient,
+} from '@/types/identity.type';
+
+export const useIdentityUserList = () => {
+  const {
+    data: userListData,
+    isLoading: isUserListLoading,
+    isError: isUserListError,
+    error: userListError,
+    isFetched,
+  } = useQuery<ApiResponse<string[]>, ApiError>({
+    queryKey: getIdentityUsersIdsQueryKey(),
+    queryFn: () => getIdentityUsersIds(),
+  });
+
+  const userDetailsQueries = useQueries({
+    queries:
+      isFetched && userListData?.data.length > 0
+        ? userListData.data.map(
+            (userId): UseQueryOptions<ApiResponse<Identity>, ApiError> => ({
+              retry: false,
+              queryKey: getIdentityUserQueryKey(userId),
+              queryFn: () => getIdentityUser(userId),
+            }),
+          )
+        : [],
+  });
+
+  const resultStatus = {
+    isLoading:
+      isUserListLoading ||
+      userDetailsQueries.some(({ isLoading }) => isLoading),
+    isError:
+      isUserListError || userDetailsQueries.some(({ isError }) => isError),
+    error:
+      userListError || userDetailsQueries.find(({ error }) => error)?.error,
+    usersInError: userDetailsQueries
+      .map(({ isError }, index) =>
+        isError ? userListData?.data?.[index] : null,
+      )
+      .filter(Boolean),
+  };
+
+  const combinedUsers: Identity[] =
+    !resultStatus.isLoading && !resultStatus.isError
+      ? userDetailsQueries.reduce((acc, { data }) => {
+          if (data) {
+            return acc.concat(data?.data);
+          }
+          return acc;
+        }, [] as Identity[])
```

Comment:
**nitpick** : `.reduce((acc, { data }) => data ? acc.concat(data?.data) : acc, [] as Identity[])` 
Could be applied to `combinedGroups` & `combinedServiceAccounts`

(more concise but it would be a ternary inside a ternary)